Common: changed civetweb line in rgw section(conf)

SirishaGuduru opened the pull request:
Resolves issue: Multiple RGW Ceph.conf Issue #1258

In multi-RGW setup, in ceph.conf the RGW sections
contain identical bind IP in civetweb line. So this
modification fixes that issue and puts the right IP
for each RGW.

Closes: https://github.com/ceph/ceph-ansible/issues/1767
Signed-off-by: SirishaGuduru SGuduru@walmartlabs.com

leseb commented:
jenkins test this please

SirishaGuduru commented:
@leseb, In this section "https://github.com/SirishaGuduru/ceph-ansible/blob/341af20e6820a924df11cb983aaa674a07954606/roles/ceph-common/templates/ceph.conf.j2#L67-L73" monitor address is basically the addresses of mon nodes. If we have RGWs residing on different nodes other than mon nodes, how will this work? And this condition gives the same IP for all RGW sections.

leseb commented:
@SirishaGuduru sorry I might have been not very clear. You really need to re-use this entire block https://github.com/SirishaGuduru/ceph-ansible/blob/341af20e6820a924df11cb983aaa674a07954606/roles/ceph-common/templates/ceph.conf.j2#L67-L80, leave the block thing **but** you have to create a set of variables for rgw:

* `raadosgw_interface`
* `radosgw_address`

Then, you need to make these variables mandatory by adding check in the rgw role, see examples from the mon here: https://github.com/ceph/ceph-ansible/blob/149f21b2eff8d37c979148fb838bb73bfd5b00a6/roles/ceph-mon/tasks/check_mandatory_vars.yml

Hope it's clearer now.
Thanks!

SirishaGuduru commented:
@leseb, its clear now. I will look at how mon is implemented and try doing the same.

leseb commented:
@SirishaGuduru thanks!

SirishaGuduru commented:
@leseb , I did try rebasing master. There was a merge conflict with this file on my branch. Hence had to do a commit.

leseb commented:
@SirishaGuduru if there is a conflict you have to resolve it then you can pursue you rebase and then push cleanly.

guits commented:
@SirishaGuduru I think you forgot to modify the `roles/ceph-defaults/defaults/main.yml`.

You shouldn't edit `*.yml.sample` files directly.
Basically you edit roles/ceph-*/defaults/main.yml` and then you execute `generate_group_vars_sample.sh` at root.

SirishaGuduru commented:
@guits, when I run this **generate_group_vars_sample.sh**, some other sample files are also being changed. Is there a way I can just this for the changes in all.yml.sample (because my changes are only in that file)?

guits commented:
As far as I know, if you rebased well on master, you shouldn't have any other changes except yours when running `generate_group_vars_sample.sh`

SirishaGuduru commented:
@guits, I'm well on par with master, except for my commit in my branch. But still when I run **generate_group_vars_sample.sh**, I see `modified:   group_vars/all.yml.sample
	modified:   group_vars/osds.yml.sample
	modified:   group_vars/rhcs.yml.sample`
where **group_vars/all.yml.sample** is with my changes.

guits commented:
@SirishaGuduru it doesn't matter, commit the changes and push them, unless I see something wrong it should be ok.

SirishaGuduru commented:
@leseb, @guits, I have made necessary changes. Please review.

SirishaGuduru commented:
@leseb, @guits can you please review ithis?

guits commented:
jenkins test this please

guits commented:
@SirishaGuduru 
Tests need to be updated, could you update them?

SirishaGuduru commented:
@guits will do that. Do you have any specific tests that I need to target on? or should I go with how monitor_interface is used in tests?

guits commented:
@SirishaGuduru yes, reproduce the same thing which has been made for `monitor_interface`/`monitor_address` for each test, please

guits commented:
jenkins test luminous-ansible2.3-update_dmcrypt

guits commented:
jenkins test luminous-ansible2.3-docker_dmcrypt_journal_collocation

guits commented:
jenkins test luminous-ansible2.3-update_dmcrypt

guits commented:
jenkins test luminous-ansible2.3-docker_dmcrypt_journal_collocation

SirishaGuduru commented:
@guits, does this look good?

gfidente commented:
@SirishaGuduru yes this is great indeed, I pointed to this PR from issue #1767 ; I don't think it addresses setting the civetweb bind ip address for the docker container; also see PR #1773

SirishaGuduru commented:
@leseb , I did not edit this file manually. This got modified when I ran generate_group_vars_sample.sh

leseb commented on roles/ceph-rgw/tasks/check_mandatory_vars.yml:
```diff
@@ -0,0 +1,8 @@
+---
```
Having a check in `roles/ceph-common/tasks/checks/check_mandatory_vars.yml` is sufficient, so you can remove this one.

leseb commented:
jenkins test luminous-ansible2.3-bluestore_dmcrypt_journal

leseb commented:
jenkins test luminous-ansible2.3-update_docker_cluster

leseb commented:
jenkins test luminous-ansible2.3-xenial_cluster

leseb commented:
jenkins test luminous-ansible2.3-update_dmcrypt

leseb commented:
jenkins test ansible2.3-bluestore_docker_dmcrypt_journal_collocation

leseb commented:
@SirishaGuduru did you run `./generate_group_vars_sample.sh`?
Your group vars files are wrong, if I pull your branch and run it, I get the following diff: https://pastebin.com/qQKddyRM

Thanks!
